File: atcoder/abc216_d.py
# https://atcoder.jp/contests/abc216/tasks/abc216_d
# https://atcoder.jp/contests/abc216/submissions/25396173

# import sys
# it = map(int, sys.stdin.buffer.read().split())
# N = next(it)
# M = next(it)

import sys
input = sys.stdin.buffer.readline
N, M = map(int, input().split())

top = [(-1, -1)] * (N + 1)
st = []
A = list()

def add_last(i):
    last = A[i][-1]
    if top[last]==(-1, -1):
        top[last] = (i, -1)
    else:
        top[last] = [top[last][0], i]
        st.append(last)

def pop_last(i):
    A[i].pop()
    if len(A[i]): add_last(i)

for i in range(M):
    l = []
    # k = next(it)
    # for j in range(k):
        # l.append(next(it))
    k = int(input())
    l = list(map(int, input().split()))
    l.reverse()
    A.append(l)
    add_last(i)

cnt = 0
while len(st):
    col = st[-1]
    st.pop()
    x, y = top[col]
    pop_last(x)
    pop_last(y)
    cnt += 1

if cnt==N:
    print("Yes")
else:
    print("No")



# # TLE
# import sys
# input = sys.stdin.buffer.readline
# sys.setrecursionlimit(10 ** 7)

# N, M = map(int, input().split())
# stk = []
# top_set = set()
# top_idx = [-1] * (N+1)

# def pop_top(idx):
#     if len(stk[idx])==0: return
#     v = stk[idx][-1]
#     if v in top_set:
#         stk[idx].pop()
#         top_set.remove(stk[top_idx[v]].pop())
#         pop_top(idx)
#         pop_top(top_idx[v])
#     else:
#         top_set.add(v)
#         top_idx[v] = idx
#     return

# for i in range(M):
#     k = int(input())
#     a = list(map(int, (input().split())))
#     a = a[::-1]
#     stk.append(a)
#     pop_top(i)

# # print(stk)
# # print(top_set)
# # print(top_idx)

# if len(top_set):
#     ans = "No"
# else:
#     ans = "Yes"
# print(ans)

File: atcoder/abc216_d.cpp
// https://atcoder.jp/contests/abc216/tasks/abc216_d
#include<iostream>
// #include<algorithm>
// #include<string>
// #include<numeric>
// #include<vector>
// #include<map>
// #include<tuple>
#include<set>
// #include<queue>
// #include<deque>
// #include<regex>
// #include<bitset>
// #include<iomanip>
// #include<complex>
// #include<stack>
// #include<functional>

#include<bits/stdc++.h>
using namespace std;
// #include <atcoder/all>
// using namespace atcoder;
#define rep(i,n) for (int i = 0; i < (n); ++i)
#define drep(i,n) for(int i = (n-1); i >= 0; i--)
#define all(v) (v).begin(),(v).end()
#define maxs(x,y) (x = max(x,y))
#define mins(x,y) (x = min(x,y))
template <class T> bool chmax(T &a, const T &b) { if (a < b) { a = b; return 1; } return 0; }
template <class T> bool chmin(T &a, const T &b) { if (b < a) { a = b; return 1; } return 0; }
template <class T> T gcd(T a, T b) { return b ? gcd(b, a % b) : a; }
template <class T> T lcm(T a, T b) { return a/gcd(a,b)*b; }
typedef pair<int, int> P;
typedef long long ll;
const int INF = 1001001001;
const ll LINF = 1001002003004005006ll;
const ll MOD = 1e9+7;

const int MAX_N = 200005;
int n, m, k;
vector<queue<int>> stk(MAX_N);
set<int> top_set;
vector<int> top_idx(MAX_N, -1);

void check_top(int idx) {
    // cout << "run check_top" << endl;
    // for (int v: stk[idx]) cout << v << ' ';
    // for (int v: top_set) cout << v << ' ';
    if (stk[idx].empty()) return;
    int v = stk[idx].front();
    if (top_set.count(v)) {
        stk[idx].pop();
        top_set.erase(stk[top_idx[v]].front());
        stk[top_idx[v]].pop();
        check_top(idx);
        check_top(top_idx[v]);
    } else {
        top_set.emplace(v);
        top_idx[v] = idx;
    }
    return;
}


int main() {
    // cin.tie(nullptr);
    // ios::sync_with_stdio(false);
	cin >> n >> m;
    rep(i,m) {
        cin >> k;
        vector<int> a(k);
        rep(j,k) {
            cin >> a[j];
            stk[i].push(a[j]);
        }
        // reverse(a.begin(), a.end());
        check_top(i);
    }
    // rep(i,m) for (int v: stk[i]) cout << v << ' ';
    // for (int v: top_set) cout << v << ' ';
    

    if (top_set.empty()) cout << "Yes" << endl;
    else                 cout << "No"  << endl;
    // ll ans = 0;
	// cout << ans << "\n";
	return 0;
}
